js_modules/dagster-ui/packages/ui-core/src/partitions/DimensionRangeWizard.tsx

```diff
@@ -28,22 +32,106 @@ export const DimensionRangeWizard = ({
   repoAddress?: RepoAddress;
   refetch?: () => Promise<void>;
+
+  // For multi-dimensional partitions, this is set to false because filtering by failed or missing
+  // partitions only makes sense when applied across all dimensions simultaneously.
+  // This dialog is also used to backfill jobs and in that case we also set it to false.
+  showQuickSelectButtons: boolean;
 }) => {
   const isTimeseries = dimensionType === PartitionDefinitionType.TIME_WINDOW;
   const isDynamic = dimensionType === PartitionDefinitionType.DYNAMIC;
 
   const [showCreatePartition, setShowCreatePartition] = React.useState(false);
 
+  const [selectState, setSelectState] = React.useState<
+    'all' | 'failed' | 'missing' | 'failed_and_missing' | 'latest' | 'custom'
+  >('custom');
+
+  const quickSelectButtons = showQuickSelectButtons && (
+    <Box flex={{direction: 'row', gap: 8}}>
+      <JoinedButtons>
+        {isTimeseries && (
+          <ActivatableButton
+            as={Button}
+            $active={selectState === 'latest'}
+            onClick={() => {
+              setSelectState('latest');
+              setSelected(partitionKeys.slice(-1));
+            }}
+            data-testid={testId('latest-partition-button')}
+          >
+            Latest
+          </ActivatableButton>
+        )}
+        <ActivatableButton
+          as={Button}
+          $active={selectState === 'all'}
+          onClick={() => {
+            setSelectState('all');
+            setSelected(partitionKeys);
+          }}
+          data-testid={testId('all-partition-button')}
+        >
+          All
+        </ActivatableButton>
+        <ActivatableButton
+          as={Button}
+          $active={selectState === 'failed'}
+          onClick={() => {
+            setSelectState('failed');
+            setSelected(getFailedPartitions(health, partitionKeys));
+          }}
+        >
+          All failed
+        </ActivatableButton>
+        <ActivatableButton
+          as={Button}
+          $active={selectState === 'missing'}
+          onClick={() => {
+            setSelectState('missing');
+            setSelected(getMissingPartitions(health, partitionKeys));
+          }}
+        >
+          All missing
+        </ActivatableButton>
+        <ActivatableButton
+          as={Button}
+          $active={selectState === 'failed_and_missing'}
+          onClick={() => {
+            setSelectState('failed_and_missing');
+            const failedPartitions = getFailedPartitions(health, partitionKeys);
+            const missingPartitions = getMissingPartitions(health, partitionKeys);
+            setSelected(Array.from(new Set([...failedPartitions, ...missingPartitions])));
+          }}
+        >
+          All failed and missing
+        </ActivatableButton>
+        <ActivatableButton
+          as={Button}
+          $active={selectState === 'custom'}
+          onClick={() => {
+            setSelectState('custom');
```

Comment:
I like this new piece of state but if you choose one of these options and then interact with the `DimensionRangeInput` down below, I think that we should clear selectState so none of the buttons are active anymore?

Reply:
You need to select custom to interact with it